src/pyFAI/io/integration_config.py

```diff
@@ -333,3 +419,218 @@ def pop_method(self, default=None):
         else:
             raise TypeError(f"Method type {type(method)} unsupported, method={method}.")
         return method
+
+
+@mydataclass
+class WorkerConfig:
+    """Class with the configuration from the worker."""
+    application: str="worker"
+    version: int = CURRENT_VERSION
+    poni: object = None
+    nbpt_rad: int = None
+    nbpt_azim: int = None
+    unit: object = None
+    chi_discontinuity_at_0: bool=False
+    do_solid_angle: bool=True
+    polarization_factor: PolarizationDescription = None
+    normalization_factor: float=1.0
+    val_dummy: float = None
+    delta_dummy: float = None
+    correct_solid_angle: bool = True
+    dark_current: Union[str, list] = None
+    flat_field: Union[str, list] = None
+    mask_file: str = None
+    error_model: ErrorModel = None
+    method: object = None
+    opencl_device: list = None
+    azimuth_range: list = None
+    radial_range: list = None
+    integrator_class: str = "AzimuthalIntegrator"
+    integrator_method: str = None
+    extra_options: dict = None
+    monitor_name: str = None
+    OPTIONAL: ClassVar[list] = ["radial_range_min", "radial_range_max",
+                                "azimuth_range_min", "azimuth_range_max",
+                                "integrator_name", "do_poisson"]
+    GUESSED:  ClassVar[list] = ["do_2D", "do_mask", "do_dark", "do_flat", 'do_polarization',
+                                'do_dummy', "do_radial_range", 'do_azimuthal_range', "shape"]
+    def as_dict(self):
+        "Like asdict, but with possibly some more "
+        dico = asdict(self)
+        #fiddle with the object ?
+        return dico
+
+    @classmethod
+    def from_dict(cls, dico, inplace=False):
+        """Alternative constructor,
+            * Normalize the dico (i.e. upgrade to the most recent version)
+            * accepts everything which is in OPTIONAL
+
+        :param dico: dict with the config
+        :param inplace: modify the dico in place ?
+        :return: instance of the dataclass
+        """
+        if not inplace:
+            dico = copy.copy(dico)
+        normalize(dico, inplace=True)
+        to_init = {field.name:dico.pop(field.name)
+                   for field in fields(cls)
+                   if field.name in dico}
+        self = cls(**to_init)
+        for key in cls.GUESSED:
+            if key in dico:
+                dico.pop(key)
+        for key in cls.OPTIONAL:
+            if key in dico:
+                value = dico.pop(key)
+                self.__setattr__(key, value)
+        if len(dico):
+            _logger.warning("Those are the parameters which have not been converted !"+ "\n".join(f"{key}: {val}" for key,val in dico.items()))
+        return self
```

Comment:
After `normalize`, I would expect the dict to contain only valid fields (converted from the input dict) and proper warnings to be raised. So no `GUESSED` or `OPTIONAL` fields.

Added with removing `inplace`, this boils down to:

```suggestion
        kwargs = normalize(dico, inplace=False)
        return cls(**kwargs)
```

Reply:
On the one hand, you are right, but this code should be moved in `normalize` ...